CruGlobal/mpdx-react
# MPDX- 8231 - Announcements

## Description
In this PR, I add the announcements to MPDX. There are two versions of announcements, `Banners` and `Modals`; I've added both.

You will also see that I have moved the `<AlertBanner/>` into the announcements component; this is because we don't want both showing at the same time, and we need to have an authenticated session to use GraphQL, where AlertBanner does not, meaning we can show an alert on all pages if we wanted to. AlertBanner will primarily be used to show an alert on the login page.

You will see that I also have to use font awesome, but as we only want it for this component, I load it just after the initial render and then remove it on `unmount`.



## How to Test
The banners and modals should only show announcements that are in your language. See PR here: https://github.com/CruGlobal/mpdx_api/pull/2875.

_I found it easier to test on local with a local version of the API running, as I can alter the DB quickly, but test as you prefer._

You will need to alter an existing announcement (which you haven't acknowledged) or create a new one.
If you need additional reference of what it should look like, you can go to [old.stage-mpdx.org](https://old-stage.mpdx.org/login) to see how it used to look.

### Banner UI
1. Set up a banner announcement. 
2. Ensure it is showing
3. Alter the start and end dates to test; it doesn't show outside the allotted window.
4. Ensure the title and body are showing
5. Test the banner styles to ensure the banner changes colour
6. Add actions and tests they show as expected. Actions with the style icon should display an icon; the other styles should change how the button looks.
7. Clicking the close button should acknowledge the banner, and it will not reappear. (I still need to work out how to un-acknowledge it, so you might need to create a new banner.)
_The image doesn't show on the banner._

### Modal UI
1. Set up a modal announcement. 
2. Ensure it is showing
3. Alter the start date and end date to test it doesn't show outside the allotted window.
4. Ensure the title and body are showing
5. Test that the image is showing and is styled nicely
6. The announcement style shouldn't change the modal background, so you don't need to test.
7. Add actions and tests they show as expected. Actions with the style icon should display an icon; the other styles should change how the button looks.
8. Clicking the close buttons or outside the modal should acknowledge the banner, and it will not reappear. (I still need to work out how to un-acknowledge it, so you might need to create a new banner.)


### Announcement Actions

1. The `Go` action should redirect the user to another page. And then acknowledge the announcement so it doesn't show again.
2. The `AppealCreate` action should open the create appeal modal, prefill details about the end-of-year appeal, and acknowledge the announcement.
3. The `Track` action should `dispatch` the announcement args to analytics and then acknowledge the announcement.


#### This PR will fail until I merge the API PR into production https://github.com/CruGlobal/mpdx_api/pull/2875.


## Checklist:
